woocommerce/woocommerce-gutenberg-products-block
# Add `Fixed/Repeated background` to the `Featured Category` block

This PR adds the `Fixed image` and `Repeated image` media controls for the `Featured Category` block and fixes the gradient overlay.

It also includes a small refactor of the `Featured Product` block to unify how things are done in the `Featured Category` as a first step for bigger refactor.

Fixes part of https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/6236

### Testing

How to test the changes in this Pull Request:

1. Create a new page and add a `Featured Category` block.
2. On the block settings go to `Media Settings` and toggle `Fixed image` and `Repeated image` and save.
3. Check the rendered category on the frontend is correct and matches the one on the edit page.
4. Add a gradient overlay, save, and check the gradient is rendered on the frontend.
5. Edit the block again, repeat the steps above with different combinations on `Fixed image`, `Repeated image` and any other configuration.
6. Repeat steps 1-5 for the `Featured Product` block.

### Changelog

> Add the `Fixed image` and `Repeated image` media controls to the Featured Category block
